package/MDAnalysis/coordinates/TRC.py

```diff
@@ -0,0 +1,409 @@
+                        first_block = blockname
+
+                if first_block is not None:
+                    break  # First block found
+
+        #
+        # Calculate meta-data of the trajectory
+        #
+        in_positionred_block = False
+        lastline_was_timestep = False
+
+        atom_counter = 0
+        n_atoms = 0
+        frame_counter = 0
+
+        l_blockstart_offset = []
+        l_timestep_timevalues = []
+
+        with util.anyopen(self.filename) as f:
+            for line in iter(f.readline, ""):
+                #
+                # First block of frame
+                #
+                if first_block == line.strip():
+                    l_blockstart_offset.append(f.tell() - len(line))
+                    frame_counter += 1
+
+                #
+                # Timestep-Block
+                #
+                if "TIMESTEP" == line.strip():
+                    lastline_was_timestep = True
+
+                elif lastline_was_timestep is True:
+                    l_timestep_timevalues.append(float(line.split()[1]))
+                    lastline_was_timestep = False
+
+                #
+                # Coordinates-Block
+                #
+                if "POSITIONRED" == line.strip():
+                    in_positionred_block = True
+
+                if (in_positionred_block is True) and (n_atoms == 0):
+                    if len(line.split()) == 3:
+                        atom_counter += 1
+
+                if ("END" == line.strip()) and (in_positionred_block is True):
+                    n_atoms = atom_counter
+                    in_positionred_block = False
+
+        if frame_counter == 0:
+            raise ValueError(
+                "No supported blocks were found within the GROMOS trajectory!"
+            )
+
+        traj_properties["n_atoms"] = n_atoms
+        traj_properties["n_frames"] = frame_counter
+        traj_properties["l_blockstart_offset"] = l_blockstart_offset
+
+        if len(l_timestep_timevalues) >= 2:
+            traj_properties["dt"] = l_timestep_timevalues[1] - l_timestep_timevalues[0]
+        else:
+            traj_properties["dt"] = 0
+            warnings.warn(
+                "The trajectory does not contain TIMESTEP blocks!", UserWarning
+            )
+
+        return traj_properties
+
+    def _read_GROMOS11_trajectory(self):
+        frameDat = {}
+        frameDat["step"] = int(self._frame)
+        frameDat["time"] = float(0.0)
+        frameDat["positions"] = None
+        frameDat["dimensions"] = None
+        self.periodic = False
+
+        # Read the trajectory
+        f = self.trcfile
+        for line in iter(f.readline, ""):
+            if "TIMESTEP" == line.strip():
+                tmp_step, tmp_time = f.readline().split()
+                frameDat["step"] = int(tmp_step)
+                frameDat["time"] = float(tmp_time)
+
+            elif "POSITIONRED" == line.strip():
+                tmp_buf = []
```

Comment:
rather than creating a list, appending to it, then converting this to an array, I think we've found it's faster to create a np.array of the correct size, pass the strings in to this (numpy converts strings faster than Python).  You'll just have to guard against setting more than n_atoms rows of the numpy array

Reply:
I tested that out today and have seen different results (at least with the gromos11 file format).

When I pre-allocate a numpy array of size (n_atoms,3) then I have to cast strings to np.float64 every time an atom is added and I need to guard the range of the array to avoid a possible overflow-error. Both are very costly and in the end, it does not yield any performance improvement.

As it is, I fill up the list with strings and convert and check for length only once after reading. I ask to stick with the existing solution.

Overall, the reader is slow. This is likely to to the nature of the compressed, very flexible and human-readable gromos trajectories, which are costly to parse.